From 4940ee75b343be7f993c734dd8682ae791b083bb Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 24 Mar 2014 21:04:02 +0100
Subject: [PATCH 364/378] core: don't try to relabel mounts before we loaded
 the policy

(cherry picked from commit c4bfd1691f4d3e26d6d7f34dbca941e119956e8a)
(cherry picked from commit 75f801a9efc45dab52286bab678c0baa1c9a386b)
---
 src/core/mount-setup.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/core/mount-setup.c b/src/core/mount-setup.c
index 679f289..cff83d3 100644
--- a/src/core/mount-setup.c
+++ b/src/core/mount-setup.c
@@ -166,7 +166,10 @@ static int mount_one(const MountPoint *p, bool relabel) {
 
         /* The access mode here doesn't really matter too much, since
          * the mounted file system will take precedence anyway. */
-        mkdir_p_label(p->where, 0755);
+        if (relabel)
+                mkdir_p_label(p->where, 0755);
+        else
+                mkdir_p(p->where, 0755);
 
         log_debug("Mounting %s to %s of type %s with options %s.",
                   p->what,
-- 
1.8.4.5

